src/boost_histogram/_internal/hist.py

```diff
@@ -192,7 +192,7 @@ def fill(self, *args, **kwargs):
 
         threads = kwargs.pop("threads", None)
 
-        if threads is None or threads is 1:
+        if threads is None or threads == 1:
```

Comment:
If someone passes something that is not comparable to 1, this will throw an error. The value 1 is only instantiated once in CPython, so it is safe to compare with `is`.

However, if something is not compatible with a number, it probably is invalid here, so I'm okay with the change.

Reply:
> If someone passes something that is not comparable to 1, this will throw an error.

Ok, I see your reasoning, but I would argue that we want the error to happen then.

I know that numbers are only instantiated once, but using `is` in this way is not recommended. This goes against duck-typing in Python. If the user passes an object which has `__eq__` defined, it should work. `1.0 is 1` evaluates to false, but `1.0 == 1` evaluates to true.

Reply:
> `1.0 is 1` evaluates to false, but `1.0 == 1` evaluates to true

Good point, though one could argue that floats for threads should be disallowed. Numpy for example has been forcing int-like for indexing. (I am not arguing that!) However, a custom int-like object should pass!

Reply:
I agree that passing floats is a bit stupid, but you can easily get a float by accident in Python 3:
```
# ok I have 8 cores
n_cores = 8
# and I want to run threads on half of them
n_threads = n_cores / 2
# n_threads is a float now!
```

Reply:
This is incorrect in Python 3, it should be (and this works in Python 2):

```python
n_threads = n_cores // 2
```

This is now what you have to do in Python if you want to use division in an indexing expression in Numpy, as well; Numpy no longer accepts floats.

Reply:
PS: "incorrect" does not mean people will not try it.